Makefile

```diff
@@ -100,13 +105,73 @@ lint-fix: golangci-lint ## Run golangci-lint linter and perform fixes.
 vendor:
 	go mod vendor
 
+##@ eBPF
+
+# eBPF build configuration
+EBPF_DIR ?= $(ROOT)/ebpf
+EBPF_SRC_DIR ?= $(EBPF_DIR)/src
+EBPF_BUILD_DIR ?= $(EBPF_DIR)/build
+EBPF_INCLUDES ?= -I$(EBPF_DIR)/include -I/usr/include
+
+# Find all eBPF source files
+EBPF_SOURCES := $(wildcard $(EBPF_SRC_DIR)/*.bpf.c)
+EBPF_OBJECTS := $(patsubst $(EBPF_SRC_DIR)/%.bpf.c,$(EBPF_BUILD_DIR)/%.bpf.o,$(EBPF_SOURCES))
+
+# Clang flags for eBPF compilation
+CLANG ?= clang
+CLANG_FLAGS := -O2 -g -Wall -Werror -target bpf \
+	-D__TARGET_ARCH_$(GO_ARCH) \
+	-D__BPF_TRACING__ \
+	$(EBPF_INCLUDES)
+
+.PHONY: build-ebpf
+build-ebpf: ## Build all eBPF programs
+ifeq ($(shell uname -s),Darwin)
+	@echo "Detected macOS, using Docker for eBPF build..."
+	@$(MAKE) build-ebpf-docker
+else
+	@$(MAKE) build-ebpf-native
+endif
```

Comment:
instead of recursively calling make, we could do:
```make
ifneq ($(shell uname -s), Linux)
 $(info ebpf-builder=docker)
 EBPF_BUILDER=docker
else
 $(info ebpf-builder=native)
 EBPF_BUILDER=native

.PHONY: build-ebpf
build-ebpf: build-ebpf-$(EBPF_BUILDER)
```

Reply:
I used `neq Linux` instead of `eq Darwin` to have the logic be only use native if on a Linux platform, all other platforms use a Linux based docker container. This potentially handles BSD and other UNIX-like platforms